astropy/io/fits/connect.py

```diff
@@ -162,7 +163,10 @@ def read_table_fits(input, hdu=None):
 
     # TODO: deal properly with unsigned integers
 
-    for key, value, comment in table.header.cards:
+    # Read time coordinates as Time
+    hdr = FITS_time().read_time(table.header, t)
```

Comment:
I understand the motivation, but unless I missed it I don't see where you keep and reuse a `FITS_time` object (for the time state) ? Which means that this object is not really needed.